### PR TITLE
fix: path construction errors on Windows

### LINKED ISSUE #18: lintrunner panics on Windows
Running `lintrunner` or `lintrunner init` fails with "The filename, directory name, or volume label syntax is incorrect. (os error 123)"

cargo test on Windows:

```
Error: Failed to find file while gathering files to lint: \\?\C:\Users\justinchu\AppData\Local\Temp\.tmpG29xS4/test_3.txt

Caused by:
    The filename, directory name, or volume label syntax is incorrect. (os error 123)
test git::tests::doesnt_detect_unchanged ... FAILED
Error: Failed to find file while gathering files to lint: \\?\C:\Users\justinchu\AppData\Local\Temp\.tmpziOhRn/new.txt
```

### PR DESCRIPTION
**Pull request by justinchuby**
- Fix a path construction issue in git.rs and another one in persistent_data.rs.
- Run CI test on windows and macos.
- Skip some tests on windows

fixes #18

### CODE REVIEW

**Comment by justinchuby** [comment]
I can't seem to figure out the integration tests

**Comment by suo** [comment]
the integration tests are snapshotted against linux output—looks like windows gives slightly different error messages

**Comment by justinchuby** [comment]
There are error code 123 file path invalid errors: I suspect path construction is breaking somewhere

**Comment by justinchuby** [comment]
Looks like its happening at

https://github.com/suo/lintrunner/blob/1c991affb15edac2bb67080e49bf0e5037b47e92/src/path.rs#L44-L47

an input

```
"p = C:\\Users\\JUSTIN~1\\AppData\\Local\\Temp\\test-lintrunner-configPewVvf.toml"
```

was turned into

```
"canonicalize = \\\\?\\C:\\Users\\justinchu\\AppData\\Local\\Temp\\test-lintrunner-configPewVvf.toml"
```

which then fails

https://github.com/suo/lintrunner/blob/1c991affb15edac2bb67080e49bf0e5037b47e92/src/main.rs#L155

**Comment by justinchuby** [comment]
```
"cur_run_dir: \"C:\\\\Users\\\\justinchu\\\\AppData\\\\Roaming\\\\lintrunner\\\\data\\\\e8d5572c7f8a113744a9cc4430ab30c8c7c5382a8459383a6b8e227d4b198a7e\\\\runs\\\\2022-11-16T09:39:09.794-08:00_e229899eed90c4abf0ad1da8bbdfaada91890b91966337def78ab9dad2fea417\""
```

this is an invalid dir in windows

File: tests/integration_test.rs [hunk 1]
```diff
@@ -263,6 +263,7 @@ fn linter_providing_nonexistent_path_degrades_gracefully() -> Result<()> {
 }
```
Comment (justinchuby) [inline tests/integration_test.rs:263]: Need to fix echo on windows too

**Comment by justinchuby** [comment]
@suo I think I fixed it. Please take a look. Thanks!

**Comment by suo** [comment]
LGTM, thanks! I'd love to get the windows tests enabled at some point but it shouldn't block for now.

**Pull request merged**